feature-wallet-impl/src/main/java/io/novafoundation/nova/feature_wallet_impl/data/network/blockchain/assets/transfers/BaseAssetTransfers.kt

```diff
@@ -48,16 +48,20 @@ abstract class BaseAssetTransfers(
      */
     protected abstract suspend fun transferFunctions(chainAsset: Chain.Asset): List<Pair<String, String>>
 
-    override suspend fun performTransfer(transfer: WeightedAssetTransfer): Result<ExtrinsicSubmission> {
-        return extrinsicService.submitExtrinsic(transfer.originChain, transfer.sender.intoOrigin()) {
-            transfer(transfer)
-        }
+    override suspend fun performTransfer(transfer: WeightedAssetTransfer, coroutineScope: CoroutineScope): Result<ExtrinsicSubmission> {
+        return extrinsicServiceFactory
+            .create(coroutineScope)
+            .submitExtrinsic(transfer.originChain, transfer.sender.intoOrigin()) {
+                transfer(transfer)
+            }
     }
 
-    override suspend fun calculateFee(transfer: AssetTransfer): Fee {
-        return extrinsicService.estimateFee(transfer.originChain, TransactionOrigin.SelectedWallet) {
-            transfer(transfer)
-        }
+    override suspend fun calculateFee(transfer: AssetTransfer, coroutineScope: CoroutineScope): Fee {
+        return extrinsicServiceFactory
+            .create(coroutineScope)
+            .estimateFee(transfer.originChain, TransactionOrigin.SelectedWallet) {
```

Comment:
Can you please point me where do we pass FeePayment.Asset based on the `transfer`? Also I havent seen a modification of FeeLoaderMixin to account for fee selector, is it going to be a separate PR?